### PR TITLE
fix(compiler-cli): resolve resource URLs before loading them under en…

### PR DESCRIPTION
**Pull request by alexeagle**
…ableResourceInlining

Also turn on the feature for Bazel ng_module rules

### CODE REVIEW

**Review by alxhub — APPROVED** [review]

File: packages/compiler-cli/src/transformers/inline_resources.ts [hunk 1]
```diff
@@ -12,70 +12,85 @@ import {MetadataObject, MetadataValue, isClassMetadata, isMetadataImportedSymbol
...
-          if (style) return style;
-          allStylesInlined = false;
-          return styleUrl;
-        });
-        if (allStylesInlined) {
-          arg['styles'] = newStyles;
-          delete arg.styleUrls;
-        }
+
+    const styles = arg['styles'];
```
Comment (alxhub) [inline packages/compiler-cli/src/transformers/inline_resources.ts]: Can simplify with `|| []`

**Review by alxhub — APPROVED** [review]

**Review by alxhub — APPROVED** [review]

File: packages/compiler-cli/src/transformers/inline_resources.ts [hunk 2]
```diff
@@ -254,53 +266,59 @@ function updateComponentProperties(
...
-                    node, ts.createIdentifier('template'), ts.createLiteral(template));
-              }
-            }
-            return node;
-
-          default:
-            return node;
+      case 'styleUrls':
+        if (!ts.isArrayLiteralExpression(prop.initializer)) {
+          throw new Error('styles takes an array argument');
```
Comment (alfaproject) [inline packages/compiler-cli/src/transformers/inline_resources.ts]: `styleUrls`

File: packages/compiler-cli/src/transformers/inline_resources.ts [hunk 3]
```diff
@@ -254,53 +266,59 @@ function updateComponentProperties(
...
+          const styles = loader.get(expr.text);
+          return ts.createLiteral(styles);
+        }));
+        break;
+
+      case 'templateUrl':
+        if (!ts.isStringLiteral(prop.initializer) &&
+            !ts.isNoSubstitutionTemplateLiteral(prop.initializer)) {
+          throw new Error(
+              'Can only accept string literal arguments to templateUrl. ' + PRECONDITIONS_TEXT);
```
Comment (alfaproject) [inline packages/compiler-cli/src/transformers/inline_resources.ts]: `templateUrl` is not an array, so maybe reword this?

File: packages/compiler-cli/src/transformers/inline_resources.ts (hunk 2 above)
Comment (alexeagle) [reply packages/compiler-cli/src/transformers/inline_resources.ts]: done

File: packages/compiler-cli/src/transformers/inline_resources.ts (hunk 3 above)
Comment (alexeagle) [reply packages/compiler-cli/src/transformers/inline_resources.ts]: done
